Detectors/MUON/MCH/DevIO/Digits/DigitIOV0.cxx

```diff
@@ -28,73 +28,48 @@ namespace o2::mch::io::impl
 {
 void DigitReaderV0::count(std::istream& in, size_t& ntfs, size_t& nrofs, size_t& ndigits)
 {
-  if (mCountDone) {
-    return;
-  }
-
   rewind(in);
-  mROFs.clear();
-  mTFs.clear();
   ndigits = 0;
+  nrofs = 0;
   ntfs = 0;
-  size_t index = 0;
-  uint32_t orbit = 0;
-  size_t pos = in.tellg();
   int ndig{0};
-  nrofs = 0;
-
   auto dff = digitFileFormats[0];
 
   while ((ndig = advance(in, dff.digitSize, "digits")) >= 0) {
-    ++nrofs;
-    int next = in.tellg();
-    mROFs.emplace_back(pos, next - 1);
-    mTFs.emplace_back(mROFs.size() - 1, mROFs.size() - 1);
-    index += ndig;
     ndigits += ndig;
-    pos = next;
-    ++orbit;
+    ++nrofs;
+    ++ntfs;
   }
-  nrofs = mROFs.size();
-  ntfs = mTFs.size();
   rewind(in);
-  mCountDone = true;
 }
 
 bool DigitReaderV0::read(std::istream& in,
                          std::vector<Digit>& digits,
                          std::vector<ROFRecord>& rofs)
 {
-  if (!mCountDone) {
-    size_t ntfs, nrofs, ndigits;
-    count(in, ntfs, nrofs, ndigits);
-  }
+  digits.clear();
```

Comment:
@pillot will have a closer look tomorrow to all your changes (thanks!) but I think the clear is actually not needed in both implementations as it is done by the DigitReader::read itself ?

Reply:
I agree it is redundant. I added it here as a protection because the current implementations can only work with empty vectors. Appending digits and rofs to non-empty vectors, although possible, would require to change the digit references in the rofs (like it is done in digits-file-reader-workflow). But we can also consider it as a prerequisite to use these functions and remove the clear().